### PR TITLE
update tests

### PR DESCRIPTION
**Pull request by math411**
*Issue #, if available:*

*Description of changes:*

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ ] I have read the [CONTRIBUTING](https://github.com/amazon-braket/amazon-braket-simulator-v2-python/blob/main/CONTRIBUTING.md) doc
- [ ] I used the commit message format described in [CONTRIBUTING](https://github.com/amazon-braket/amazon-braket-simulator-v2-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/amazon-braket/amazon-braket-simulator-v2-python/blob/main/README.md) and [API docs](https://github.com/amazon-braket/amazon-braket-simulator-v2-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by kshyatt-aws — APPROVED** [review]

**Comment by math411** [comment]
`PYTHON_JULIACALL_OPTIMIZE` brought down local runtime from `~2:40` to `~2:00`

**Comment by kshyatt-aws** [comment]
What level? 3?

File: requirements-test.txt [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+isort
+pre-commit
+pylint
+pytest==7.1.2
+pytest-benchmark
+pytest-cov
+pytest-rerunfailures
+pytest-timeout
+pytest-xdist
+qiskit==1.1.2
```
Comment (kshyatt-aws) [inline requirements-test.txt:14]: Would be good to set this to the current NBI version of qiskit (1.0.2?)

Comment (math411) [reply requirements-test.txt:14]: I am looking to get that bumped up in https://github.com/amazon-braket/amazon-braket-examples/pull/614. I will adjust to use `>=` accordingly

**Comment by math411** [comment]
> What level? 3?

That had the worst performance. It seemed `0` resulted in the best. I couldn't find docs on the highest vs lowest so I benchmarked across the levels.

**Comment by kshyatt-aws** [comment]
https://docs.julialang.org/en/v1/manual/command-line-interface/#command-line-interface they're the same levels as here

**Comment by kshyatt-aws** [comment]
Generally higher is supposed to be more performant

**Review by kshyatt-aws — APPROVED** [review]

**Pull request merged**